### PR TITLE
Narrative indexer updates

### PR DESCRIPTION
**Pull request by jayrbolton**
This is the last in a recent series of PRs

### CODE REVIEW

**Comment by jayrbolton** [comment]
Will add the test code in a minute

File: src/index_runner/es_indexers/indexer_utils.py [hunk 1]
```diff
@@ -77,35 +77,6 @@ def get_shared_users(ws_id):
     return shared_users
 
 
-def fetch_objects_in_workspace(ws_id, include_narrative=False):
```
Comment (jayrbolton) [inline src/index_runner/es_indexers/indexer_utils.py:80]: This is only used by the narrative indexer and is very specific to that, so I moved it inside that module

File: src/index_runner/es_indexers/narrative.py [hunk 2]
```diff
@@ -24,16 +26,56 @@ def index_narrative(obj_data, ws_info, obj_data_v1):
...
     # Get all the types and names of objects in the narrative's workspace.
-    narrative_data_objects = indexer_utils.fetch_objects_in_workspace(ws_id)
+    narrative_data_objects = _fetch_objects_in_workspace(ws_id)
+    # Extract all the data we want to index from the notebook cells
+    raw_cells = obj_data['data'].get('cells', [])
+    index_cells = _extract_cells(raw_cells, ws_id)
+    result = {
+        '_action': 'index',
+        'doc': {
+            'narrative_title': narrative_title,
```
Comment (jayrbolton) [inline src/index_runner/es_indexers/narrative.py:53]: New fields are: is_temporary, is_narratorial, owner, modified_at

Now that I look at it, I'm not sure why we added "modified_at" when we already have that timestamp in the generic fields (global_mappings.ws_object.timestamp). I should have caught this before.

File: src/index_runner/es_indexers/narrative.py [hunk 3]
```diff
@@ -104,3 +132,19 @@ def _process_code_cell(cell):
         index_cell['cell_type'] = 'code_cell'
         index_cell['desc'] = cell.get('source', '')
     return index_cell
+
+
+def _fetch_objects_in_workspace(ws_id):
```
Comment (jayrbolton) [inline src/index_runner/es_indexers/narrative.py:137]: Moved here from the utils module

File: src/test/mock_kbase_services/workspace_list_objects_33192.json [hunk 4]
```diff
@@ -5,13 +5,222 @@
   "body": {
```
Comment (jayrbolton) [inline src/test/mock_kbase_services/workspace_list_objects_33192.json:5]: Cached workspace response for testing narrative indexer

File: src/index_runner/es_indexers/narrative.py (hunk 2 above)
Comment (slebras) [reply src/index_runner/es_indexers/narrative.py:53]: is the plan to change before this gets merged?

**Pull request merged**